packages/services/src/state_pruner.rs

```diff
@@ -0,0 +1,261 @@
+pub mod service {
+    use metrics::{
+        prometheus::{core::Collector, IntGauge},
+        RegistersMetrics,
+    };
+
+    use crate::{Result, Runner};
+    use std::time::Duration;
+
+    use super::create_int_gauge;
+
+    pub struct StatePruner<Db, Clock> {
+        storage: Db,
+        clock: Clock,
+        retention: Duration,
+        metrics: Metrics,
+    }
+
+    impl<Db, Clock> StatePruner<Db, Clock> {
+        pub fn new(storage: Db, clock: Clock, retention: Duration) -> Self {
+            Self {
+                storage,
+                clock,
+                retention,
+                metrics: Metrics::default(),
+            }
+        }
+    }
+
+    impl<Db, Clock> StatePruner<Db, Clock>
+    where
+        Db: crate::state_pruner::port::Storage,
+        Clock: crate::state_pruner::port::Clock,
+    {
+        pub async fn prune(&self) -> Result<()> {
+            let pruned = self
+                .storage
+                .prune_entries_older_than(self.clock.now() - self.retention)
+                .await?;
+
+            let table_sizes = self.storage.table_sizes().await?;
+
+            self.metrics.observe_pruned(&pruned);
+            self.metrics.observe_table_sizes(&table_sizes);
+
+            tracing::info!("Pruned: {pruned:?}");
+            tracing::info!("Table sizes: {table_sizes:?}");
+
+            Ok(())
+        }
+    }
+
+    impl<Db, Clock> Runner for StatePruner<Db, Clock>
+    where
+        Db: crate::state_pruner::port::Storage + Send + Sync,
+        Clock: crate::state_pruner::port::Clock + Send + Sync,
+    {
+        async fn run(&mut self) -> Result<()> {
+            self.prune().await
+        }
+    }
+
+    #[derive(Clone)]
+    struct Pruned {
+        blob_transactions: IntGauge,
```

Comment:
I'm assuming the intent behind this metric is to show when pruning happened and what effect it had in terms of deleted items. 

If so consider one of two things:
1. not having this metric and using the total size of the table to get a accurate enough measurement of how much was deleted
2. if you want this metric consider making it a counter not a gauge, since the number of deleted rows is not a value that can ever decrease, it only increases. We can apply `rate` and similar functions over on Prometheus to get how much the number of rows increased in a given time period.

Reply:
Went with option 1.